tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/cli/DetachHistory.java

```diff
@@ -0,0 +1,170 @@
+ */
+package org.projectnessie.tools.contentgenerator.cli;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import org.projectnessie.client.api.NessieApiV2;
+import org.projectnessie.error.BaseNessieClientServerException;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.error.NessieReferenceConflictException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.Reference;
+import org.projectnessie.model.Tag;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+
+@Command(
+    name = "detach-history",
+    mixinStandardHelpOptions = true,
+    description =
+        "Get the latest contents from one or all branches, commit it directly on top of the root commit (no history), "
+            + "then reassign the original branch(es) to the new commit.")
+public class DetachHistory extends RefreshContent {
+
+  @CommandLine.Option(
+      names = {"--src-suffix"},
+      description =
+          "Name suffix for tagging the original (source) branches, defaults to '-original'.")
+  private String origSuffix = "-original";
+
+  @CommandLine.Option(
+      names = {"--final-suffix"},
+      description =
+          "Name suffix for tagging the the final stat of source branches (equivalent to the tag generated "
+              + "from the --src-suffix option). This suffix is used during retries, defaults to '-completed'.")
+  private String completedSuffix = "-completed";
+
+  @CommandLine.Option(
+      names = {"--max-attempts"},
+      description =
+          "Max attempts (in case of concurrent commits from other actors), defaults to 100.")
+  private int maxAttempts = 100;
+
+  private final Map<String, Branch> sources = new HashMap<>();
+  private final Map<String, Branch> targets = new HashMap<>();
+  private String tmpSuffix;
+  private String rootHash;
+
+  @Override
+  public void execute() throws BaseNessieClientServerException {
+    try (NessieApiV2 api = createNessieApiInstance()) {
+      rootHash = api.getConfig().getNoAncestorHash();
+      for (int attemt = 0; attemt < maxAttempts; attemt++) {
+        spec.commandLine().getOut().printf("Running attempt %d...%n", attemt);
+
+        sources.clear();
+        targets.clear();
+        tmpSuffix = "-tmp-" + UUID.randomUUID();
+
+        super.execute(); // copy content into temporary branches
+
+        try {
+          // reassign original branch names to HEADs of corresponding temporary branches
+          reassign(api);
+          break;
+        } catch (NessieReferenceConflictException e) {
+          spec.commandLine()
+              .getOut()
+              .printf("Unable to complete attempt %d, retrying...%n", attemt);
+        }
+      }
+    }
+    spec.commandLine().getOut().printf("Completed successfully%n");
```

Comment:
Does this print even if `maxAttempts` is exceeded?  If so and you decide to change it, then perhaps it's worth renaming attemt -> attempt in the process.

Reply:
thx - fixed